### ISSUE TITLE
Using JSONP (with ItemFileReadStore and xhrScriptPlugin) fills console with errors

### ISSUE THREAD

**Issue by aerwin** [maqetta/maqetta#2910]
@jhpedemonte @peller 

This doesn't really seem to have a functional impact, but I don't like to see developer console flooded with infinite messages. Luckily, Chrome groups messages or it would be much worse. I don't see this behavior on Maqetta.org with M6, so suspecting perhaps a Dojo 1.8 issue? The errors and warnings all kind of seem to be from the bowels of Dojo.
1. New HTML page
2. Drag GridX onto page
3. Choose "Data from URL (JSONP)"
4. Enter URL (either http://maqettaservice.appspot.com/sampleservice or http://maqetta.org/sampleService)
5. Click Finish and console starts flooding with warning messages (after a couple errors) 

One warning message:

```
Resource interpreted as Script but transferred with MIME type text/html: "http://maqettaservice.appspot.com/sampleservice?callback=parent.dojo.io.script.jsonp_dojoIoScript1._jsonpCallback". lib/dojo/dojo/request/script.js:36
attach lib/dojo/dojo/request/script.js:36
dojox.io.scriptFrame._loaded lib/dojo/dojox/io/scriptFrame.js:42
onload
```

Then, 4 of these "Uncaught: TypeErrror" errors:

```
Uncaught TypeError: Object frame0.22868602629750967 has no method 'getElementById' lib/dojo/dojo/dom.js:54
dom.byId lib/dojo/dojo/dom.js:54
remove lib/dojo/dojo/request/script.js:40
deadScripts lib/dojo/dojo/request/script.js:76
array.forEach lib/dojo/dojo/_base/array.js:244
isValid lib/dojo/dojo/request/script.js:75
watchInFlight
```

Then, ever increasing total of these (1355 and counting):

```
Resource interpreted as Script but transferred with MIME type text/html: "http://maqettaservice.appspot.com/sampleservice?callback=parent.dojo.io.script.jsonp_dojoIoScript2._jsonpCallback". lib/dojo/dojo/request/script.js:36
attach lib/dojo/dojo/request/script.js:36
dojox.io.scriptFrame._loaded lib/dojo/dojox/io/scriptFrame.js:42
onload
```

---

Then, if we save and reopen the file, the console floods with actual error messages (not the warnings from above):
1. Save and close file
2. Reopen file
3. Reopen file and "Uncaught TypeError:"'s just keep building up:

```
Uncaught TypeError: Object frame0.6635820136871189 has no method 'getElementById' lib/dojo/dojo/dom.js:54
dom.byId lib/dojo/dojo/dom.js:54
remove lib/dojo/dojo/request/script.js:40
deadScripts lib/dojo/dojo/request/script.js:76
array.forEach lib/dojo/dojo/_base/array.js:244
isValid lib/dojo/dojo/request/script.js:75
watchInFlight lib/dojo/dojo/request/watch.js:25
```

**Comment by jhpedemonte** [comment]
Is this any better after @peller  upgraded to Dojo 1.8 final?

**Comment by aerwin** [comment]
I was running with the very latest when I noticed the problem.

**Comment by jhpedemonte** [comment]
-> @aerwin, to investigate further.

**Comment by aerwin** [comment]
@jhpedemonte Do you know anything about where the code for these sample services lives, how they are deployed, etc.?

One of the messages is "Resource interpreted as Script but transferred with MIME type text/html" and no doubt looking at the response headers that's true. I'm not sure why the base dojo code as decided in 1.8 to pump out the same message over and over, but us fixing the content type is probably a big step in the right direction.

Response headers:

```
HTTP/1.1 200 OK
Date: Fri, 24 Aug 2012 00:00:32 GMT
Content-Type: text/html
Server: Google Frontend
Content-Length: 347
```

Response content:

```
parent.dojo.io.script.jsonp_dojoIoScript7._jsonpCallback({ identifier: 'id',    label: 'name',  items: [        { id: '0', name:'Foods', numberOfItems:1},      { id: '1', name:'<b>Fruits</b>', numberOfItems:1},      { id: '4' ,name:'Citrus', numberOfItems:1},         { id: '2', name:'Vegetables', numberOfItems:0},     { id: '3', name:'Cereals', numberOfItems:0} ] } )
```

**Comment by jhpedemonte** [comment]
Well, one is on maqetta.org, so we'd need to properly set up the mime type for that file.

The one on appspot was uploaded by @billreed63, correct?  Not sure how that works or how to edit the mime type.

We can look to fix this issue, but we should also open a bug on Dojo to not output an endless amount of warnings/errors.

**Comment by aerwin** [comment]
I've looked at this a bit more since my last post... 

I think I can open a legitimate Dojo issue against the `Uncaught TypeError:`'s because as I've explored this some more since my last post I notice they happen when I preview the page outside of the visual editor.

But, the unlimited MIME type errors only seem to occur when we first create a JSONP-based widget (not just GridX) in the visual editor. So, I don't think I can give the Dojo team a reproducible test case for that piece of it. Fixing the MIME type is probably our best bet to git rid of those.

**Comment by aerwin** [comment]
Opened Dojo ticket for the uncaught type error's:
- Using JSONP (with ItemFileReadStore and xhrScriptPlugin) fills console with errors (http://bugs.dojotoolkit.org/ticket/15890)

**Comment by aerwin** [comment]
@jhpedemonte @JonFerraiolo 

-> M8
-> Change title to "Using JSONP (with ItemFileReadStore and xhrScriptPlugin) fills console with errors"
-> Spin-off mime-type portion to #3075

The Dojo team has classified the ticket I opened for the uncaught type errors as a regression and assigned a target release of 1.8.1.

**Comment by peller** [comment]
should be fixed in Dojo 1.8.1
